### PR TITLE
Fix issue with search in path cache.

### PR DESCRIPTION
**Pull request by AGulev**
Since we don't work with path themselves but with strings, the logic that search in cache strings should take into account corner cases like it isn't enough to check `.startwith` path `test` when you are checking `test_folder/path.file`.

Tests also added

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/BundleResourcesTest.java [hunk 1]
```diff
@@ -191,7 +191,6 @@ public void testcollectBundleResources() throws Exception {
         assertTrue(resourceMap.containsKey("test.txt"));
 
         project.getProjectProperties().putStringValue("project", "bundle_resources", "/restest2");
-        project.cleanupResourcePathsCache();
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/BundleResourcesTest.java:194]: it's not needed here

File: com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/test/ProjectTest.java [hunk 2]
```diff
@@ -122,6 +122,7 @@ public void setUp() throws Exception {
         libraryUrls.add(new URL("http://localhost:8081/test_lib2.zip"));
         libraryUrls.add(new URL("http://" + BASIC_AUTH + "@localhost:8081/test_lib5.zip"));
         libraryUrls.add(new URL("http://" + BASIC_AUTH_ENV_TOKEN + "@localhost:8081/test_lib6.zip"));
+        libraryUrls.add(new URL("http://localhost:8081/test.zip"));
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/test/ProjectTest.java:125]: I added new mount for my test

File: com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/test/ProjectTest.java [hunk 3]
```diff
@@ -284,6 +285,30 @@ public void testFindResourceDirs() throws Exception {
...
+        assertEquals(0, pathResults.size());
+
+        ArrayList<String> pathResults1 = new ArrayList<String>();
+        project.findResourcePaths("/test/file.in", pathResults1);
+        assertEquals(1, pathResults1.size());
+
+        ArrayList<String> pathResults2 = new ArrayList<String>();
+        project.findResourcePaths("/test", pathResults2);
+        assertEquals(1, pathResults2.size());
+    }
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/test/ProjectTest.java:310]: check that test situations like `test` isn't `test_1/`

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java [hunk 4]
```diff
@@ -2340,13 +2334,25 @@ public void handleFile(String filePath, Collection<String> results) {
...
+        if ((ext == null || ext.isEmpty()) && (_path == null || _path.isEmpty())) {
             result.addAll(allResourcePathsCache);
             TimeProfiler.stop();
             return;
         }
+        IResource res = fileSystem.get(_path);
+        try {
+            if ( (_path != null && !_path.isEmpty()) && (!res.isFile() || res.getContent() == null)) {
+                if (!_path.endsWith("/")) {
+                    _path += "/";
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java:2349]: actual fix

Comment (JCash) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java:2349]: Trailing `/` makes sense when it's not a file (i.e. dir), but when it's an empty file, do we get an empty byte array, or null? (I don't remember)

**Review by JCash — APPROVED** [review]

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java (hunk 4 above)
Comment (AGulev) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java:2349]: It will be an empty byte array. 
I added this because of how MockFileSystem works: returns true for a folder when call `isFile()`
I wanted to fix MockFS, but then decided to add another check in cases some other filesystem works the same. In most of the cases this condition will not even be checked
